src/base/image_reader.cc

```diff
@@ -181,15 +181,30 @@ ImageReader::Status ImageReader::Next(Camera* camera, Image* image,
       return Status::CAMERA_SINGLE_DIM_ERROR;
     }
 
+    //////////////////////////////////////////////////////////////////////////////
+    // Read camera model and check for consistency if it exists
+    //////////////////////////////////////////////////////////////////////////////
+    std::string camera_model = "Undefined";
+    bitmap->ExifCameraModel(&camera_model);
```

Comment:
If I read this code here correctly, then this would always select a single camera, if the `camera_model == "Undefined"`? This doesn't seem right.

Reply:
We should also check whether we successfully retrieved the camera_model by checking the return value of ExifCamerModel

Reply:
Due to the "hashing" by make, model, focal length, and width+height the call to `ExifCameraModel` will always return something meaningful even if it's just the "WidthxHeight" descriptor. Thus, even for missing EXIF tags it will still only group together images with the same dimensions (which I think is fine in this case) and won't force a single camera for the whole model.

Returning false from `ExifCameraModel` when an EXIF tag is missing will break the setup since the `cameras_` is a map indexed by `camera_model` so we do want this to always succeed for the `AUTO` mode to work correctly. Returning an empty string would essentially be forcing the use of a single camera model in the `AUTO` mode when EXIF tags are missing. But, maybe that's a more acceptable behavior; I'm open to handle this in whichever way makes most sense to you.

Reply:
Okay, so correct me if I am wrong, but the way I understand the current behavior: if only width_height are available, then it will use the same camera for all images that only have width_height EXIF fields and where they are equal. The behavior I want is that: if any of the make/model/focal/width/height fields are missing, instantiate a new/independent camera for that image. I really only want to group them to a single camera, if all the fields are present and their values match exactly.

Reply:
OK I see your take now; I can certainly do it this way. It will take a bit of restructuring but it will get there.

Reply:
Now the logic should work correctly; tested locally and results agreed with expectation of camera models. The map used to keep track of existing camera models now gets populated only if the camera model is valid, thus cameras where the model cannot be found will not get grouped together.